File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = {}
        
        for i, n in enumerate(nums):
            diff = target - n
            if diff in res:
                return [res[diff], i]
            res[n] = i

        

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let l = 0;
        let r = nums.length - 1;

        while (l<=r) {
            const m = l + Math.floor((r - l)/2);
            if (nums[m] === target) {
                return m;
            }
            if (nums[l] <= nums[m]) {
                if (target > nums[m] || target < nums[l]) {
                    l = m + 1;
                } else r = m - 1;
            } else {
                if(nums[m] > target || target > nums[r]) {
                    r = m - 1;
                } else l = m + 1;
            }
        }
        return -1;
    }
}
